### DISCUSSION TITLE
bare repository and working directory

### DISCUSSION THREAD

**Discussion by huythanhpham138** [Schweinepriester/github-profile-achievements#63] [Q&A]
What is the difference between "bare repository" and "working directory"?

**Comment by melvincwng** [comment]
@huythanh13082000, a 'bare repository' essentially is a git repository whereby you cannot make changes to your files (e.g. add/edit files) and cannot track things like your commit history. Essentially it's pretty much useless nowdays. The reason for this is due to a lack of working directory present in your git repo when you do git init --bare.

In fact, nowdays when you do 'git init', you are essentially creating a default or non-bare git repository, which does have the working directory or folder structure required to add new files, edit files or track commit history.

A 'working tree' or 'working directory', on the other hand, is a directory in which all your project files/sub-directories reside. In layman terms, you can view it as the structure of your files in the codebase.

Hope I have simplified the concepts of you.

For more information, please refer to this [reference](https://www.geeksforgeeks.org/bare-repositories-in-git/amp/) and this [reference](https://stackoverflow.com/questions/7861184/what-is-the-difference-between-git-init-and-git-init-bare#:~:text=git%20init%20creates%20a%20git,not%20have%20the%20working%20directory.).

If you find this helpful, please kindly 'Mark as answer'.

Thank you and have a nice day 😊
